src/tweenable.rs

```diff
@@ -404,7 +420,11 @@ impl<T> Tweenable<T> for Tween<T> {
     }
 
     fn is_looping(&self) -> bool {
-        self.tweening_type != TweeningType::Once
+        match self.count {
+            RepeatCount::Finite(times) if times == 1 => false,
+            RepeatCount::Finite(times) => self.times_completed < times,
+            RepeatCount::Infinite => true,
+        }
```

Comment:
I kept this from the original PR, but I'm not sure if it's the best approach. Currently, there's no way to seek to a specific number of completions: the AnimClock always treats progress as within the 0-1 range of a single run. We could use the truncated part to represent the number of completions, but there are only 23 mantissa bits and we support 32 bits of completion so that doesn't really work.

I'm currently think all the `is_looping` stuff should be removed and an internal `AnimCompletion` class should be created to track when the overall animation is completed (while keeping `AnimClock` in charge of an individual run's completions). Later if there's demand, an API could be added to `Tweenable` that lets you set the number of completions. WDYT?

Reply:
Note that this would mean set_progress always uses `fract` instead of clamping.

Reply:
> We could use the truncated part to represent the number of completions, but there are only 23 mantissa bits and we support 32 bits of completion so that doesn't really work.

I don't think that's a problem. I have a hard time imagining anyone requesting an animation to repeat exactly 2^26 times, not one more or less. But this raises another issue, which is that the larger the number the less precision in the fractional part (that's how floats work) and so `set_progress()` as is, taking into account the integral part, is not ideal.

Instead I think we should move to `set_progress()` being in [0:1] always (or even [0:1[ if looping), and provide another way to set the times completed. In particular, it might make sense to directly allow setting the elapsed time, which is encoded as `u128` (via `Duration`) so does not suffer from variable precision. Thinking way ahead of time, that would also allow integrating with a timeline UI in some Editor.

> I'm currently think all the `is_looping` stuff should be removed and an internal `AnimCompletion` class should be created to track when the overall animation is completed (while keeping `AnimClock` in charge of an individual run's completions).

I'm not sure I see what that looks like. The point of `AnimClock` is that it should make it easier to handle the case where `tick()` crosses one (or, less likely, several) loop boundary/ies. I'm not sure what `AnimCompletion` would provide beyond what `times_completed : u32` already gives us.

For the case of `is_looping()`, I think it makes few sense from the user's perspective; what makes more sense is querying the repeat count. And internally although we generally still need to know if the tweenable needs to wrap around at the end of this loop, we're not interested in whether the animation is looping, but rather whether the current loop is the last one (completed) or if there's another one after that (wrap around, still active). So either way `is_looping()` probably doesn't make sense anymore once we support a repeat count.

Reply:
Ok, I got rid of is_looping and moved towards clamp only support. For now I haven't added an API to set the completion count since I'd rather wait until a use case shows up which we can design against.

I also added AnimCompletion. The idea is just to have something keep track of when the animation is finished so that we can use it for sequences and tracks too.